packages/runtime/src/runtimes/react/index.tsx

```diff
@@ -52,34 +58,58 @@ export interface ReactRuntime {
     elementTree: ReactPage.ElementData,
     replacementContext: ReactPage.SerializableReplacementContext,
   ): ReactPage.Element
+  getBreakpoints(): Breakpoints
 }
 
-function createReactRuntime(store: ReactPage.Store): ReactRuntime {
-  return {
-    registerComponent(component, { type, label, icon = 'Cube40', hidden = false, props }) {
-      const unregisterComponent = store.dispatch(
-        registerComponentEffect(type, { label, icon, hidden }, props ?? {}),
-      )
+export class ReactRuntime {
+  static registerComponent<
+    P extends Record<string, PropControllerDescriptor>,
+    C extends ReactPage.ComponentType<{ [K in keyof P]: PropControllerDescriptorValueType<P[K]> }>,
+  >(
+    component: C,
+    {
+      type,
+      label,
+      icon = 'Cube40',
+      hidden = false,
+      props,
+    }: { type: string; label: string; icon?: ComponentIcon; hidden?: boolean; props?: P },
+  ): () => void {
+    // TODO: We're using storeContextDefaultValue here to be backward compatible.
+    // In the future we might want to create a new store, and then pass the store to StoreContext.Provider instead.
+    const unregisterComponent = storeContextDefaultValue.dispatch(
+      registerComponentEffect(type, { label, icon, hidden }, props ?? {}),
+    )
 
-      const unregisterReactComponent = store.dispatch(
-        registerReactComponentEffect(type, component as unknown as ReactPage.ComponentType),
-      )
+    const unregisterReactComponent = storeContextDefaultValue.dispatch(
+      registerReactComponentEffect(type, component as unknown as ReactPage.ComponentType),
+    )
 
-      return () => {
-        unregisterComponent()
-        unregisterReactComponent()
-      }
-    },
-    copyElementTree(
-      elementTree: ReactPage.ElementData,
-      replacementContext: ReactPage.SerializableReplacementContext,
-    ) {
-      return ReactPage.copyElementTree(store.getState(), elementTree, replacementContext)
-    },
+    return () => {
+      unregisterComponent()
+      unregisterReactComponent()
+    }
+  }
+  static copyElementTree(
+    elementTree: ReactPage.ElementData,
+    replacementContext: ReactPage.SerializableReplacementContext,
+  ): ReactPage.Element {
+    return ReactPage.copyElementTree(
+      storeContextDefaultValue.getState(),
+      elementTree,
+      replacementContext,
+    )
+  }
+  static getBreakpoints(): Breakpoints {
+    return ReactPage.getBreakpoints(storeContextDefaultValue.getState())
   }
-}
 
-export const ReactRuntime = createReactRuntime(storeContextDefaultValue)
+  constructor({ unstable_breakpoints }: { unstable_breakpoints?: BreakpointsInput } = {}) {
+    if (unstable_breakpoints) {
+      storeContextDefaultValue.dispatch(setBreakpoints(parseBreakpointsInput(unstable_breakpoints)))
+    }
+  }
```

Comment:
Constructing a runtime dispatches on the default context? We should instead return a new runtime, which its own store, and add all the static methods as instance methods to this class. What this is doing right now is _very_ confusing

Reply:
Yes, but it's kinda hard to do this right now because we're using `storeContextDefaultValue` in some places, and we want to make this backward compatible.

This is just the very minimal version so we can start using the new API. I was thinking that we can update the internal implementation later.

Reply:
We decided to not refactor the `ReactRuntime` into class for now. The current unstable public API is:
```ts
ReactRuntime.unstable_setBreakpoints({
  desktop: {
    label: "Ultra screen",
    min: 1600,
  },
  max: {
    label: "Max screen",
    min: 1200,
    max: 1599,
    viewport: 1400,
  },
  pro: {
    min: 600,
    max: 1199,
    viewport: 800,
  }
});
```